### PR TITLE
Fixed #23755 -- Added support for multiple field names in the no-cache Cache-Control directive to patch_cache_control().

### PR DESCRIPTION
**Pull request by fcurella**

### CODE REVIEW

File: tests/cache/tests.py [hunk 1]
```diff
@@ -1722,6 +1722,14 @@ def test_patch_cache_control(self):
                 parts = set(cc_delim_re.split(response['Cache-Control']))
                 self.assertEqual(parts, expected_cc)
 
+    def test_patch_cache_control_no_cache(self):
+        response = HttpResponse()
+        patch_cache_control(response, no_cache='SetCookie')
+        patch_cache_control(response, no_cache='foo')
+        self.assertEqual(response['Cache-Control'], 'no-cache=SetCookie, no-cache=foo')
+        patch_cache_control(response, no_cache=True)
+        self.assertEqual(response['Cache-Control'], 'no-cache')
```
Comment (felixxm) [inline tests/cache/tests.py]: This work without this patch, i.e. `True` takes precedence over `no-cache` values without it. Extra tests are fine, but should be added in a separate commit, I would add new tests cases to `test_patch_cache_control()`:

- 1st commit with
```python
# no-cache,
('no-cache=SetCookie', {'no_cache': True}, {'no-cache'}),
('no-cache=SetCookie,no-cache=foo', {'no_cache': True}, {'no-cache'}),
```

- 2nd commit with a new feature and:
```python
('no-cache=SetCookie', {'no_cache': 'foo'}, {'no-cache=SetCookie', 'no-cache=foo'}),
```

File: django/utils/cache.py [hunk 2]
```diff
@@ -58,12 +59,11 @@ def dictvalue(t):
...
-
+    cc = MultiValueDict()
     if response.get('Cache-Control'):
-        cc = cc_delim_re.split(response['Cache-Control'])
-        cc = dict(dictitem(el) for el in cc)
-    else:
-        cc = {}
+        fields = cc_delim_re.split(response['Cache-Control'])
+        for field in fields:
+            cc.appendlist(*dictitem(field))
```
Comment (felixxm) [inline django/utils/cache.py]: Multiple values should be allowed only for the `no-cache` directive, this change allows for multiple values for all keys what is not desirable.

File: docs/topics/cache.txt [hunk 3]
```diff
@@ -1277,12 +1277,47 @@ Here are some more examples:
...
 .. _IANA registry: https://www.iana.org/assignments/http-cache-directives/http-cache-directives.xhtml
 
+``no_cache`` is a special case in that it can be set to ``True`` or to a list of values.
+
+If you call ``cache_control(no_cache='...')`` multiple times, all the values will used in the
+resulting ``Cache-Control`` header::
+
+    from django.views.decorators.cache import cache_control, patch_cache_control
+
+    @cache_control(no_cache=['Set-Cookie', 'foo'])
```
Comment (fcurella) [inline docs/topics/cache.txt]: @felixxm I don't like using `foo` and `bar` in docs, but I couldn't find any real-world example of what values people would use, other than `Set-Cookie` as mentioned by the ticket author.

File: tests/cache/tests.py [hunk 4]
```diff
@@ -1709,6 +1709,11 @@ def test_patch_cache_control(self):
             ('must-revalidate,max-age=60,private', {'public': True}, {'must-revalidate', 'max-age=60', 'public'}),
             ('must-revalidate,max-age=60,public', {'private': True}, {'must-revalidate', 'max-age=60', 'private'}),
             ('must-revalidate,max-age=60', {'public': True}, {'must-revalidate', 'max-age=60', 'public'}),
+
+            # no-cache
+            ('no-cache=SetCookie', {'no_cache': True}, {'no-cache'}),
+            ('no-cache=SetCookie,no-cache=foo', {'no_cache': True}, {'no-cache'}),
+            ('no-cache=SetCookie', {'no_cache': 'foo'}, {'no-cache=SetCookie', 'no-cache=foo'}),
```
Comment (felixxm) [inline tests/cache/tests.py]: `SetCookie` -> `Set-Cookie`

File: docs/topics/cache.txt (hunk 3 above)
Comment (felixxm) [reply docs/topics/cache.txt]: IMO, using multiple field names in `no-cache` directive is so niche that we don't need to explain this in our documentation. I'm going to leave only `versionchanged` admonition.

**Comment by felixxm** [comment]
@fcurella Thanks :+1: I merged extra tests in 9facc9002079d2713bb2591ca01f3aca0b678a00, updated release notes and docs. I also pushed edits in 45fd6e0ed7c58b7508b5ea913d6c9ffb3af8d98d to use `defaultdict(set)` instead of `MultiValueDict`. What do you think?

**Comment by fcurella** [comment]
@felixxm I think using `set` instead of `list` for values is a good idea, it will prevent duplicate field values.

LGTM

**Comment by felixxm** [comment]
@fcurella Thanks!

**Comment by adamchainz** [comment]
Nice one!
